### DISCUSSION TITLE
Partials to variable?

### DISCUSSION THREAD

**Discussion by magick93** [hofstadter-io/hof#258] [Q&A]
Hello

Is it possible to assign the output of a partial template to a variable?

So far I've not managed to make this work, but would like to hear if this should be possible. 

Thanks

**Comment by verdverm** [comment]
I wondered this a few days ago myself! iirc, helm has support for this.

I think it's a pretty powerful technique, you could then even pass to another template or helper.

If we don't support it (which I think is probably the case), then it should be added. 

---

Created: https://github.com/hofstadter-io/hof/issues/259

**Reply by verdverm** [reply]
@magick93 out of curiosity, what are you doing where saving paritals to a template would be helpful?

I have a use case related to hof/chat and Tree of Thought, Mixture of Experts, iirc, I hit a roadblock somewhere and recall thinking I wanted this capability

**Reply by magick93** [reply]
I had a complex series of `if` statements that I needed to re-use across various partial templates. I thought if I could move these to a common template, and assign the output to a variable, it could simplify the structure.

**Reply by verdverm** [reply]
ah, interesting, I have used shared template code by passing in the name of the partial to use as a variable

I shall be interested to see both methods side-by-side after the addition

What do you think of the name template func name `render` instead of `import`?

**Reply by magick93** [reply]
> I have used shared template code by passing in the name of the partial to use as a variable

yes, I thought about doing that.

> What do you think of the name template func name render instead of import?

I dont have a a strong opinion either way..

**Reply by verdverm** [reply]
https://github.com/hofstadter-io/hof/pull/267

Merged and on tip if you'd like to try it out
